components/PreferencesButton/index.js

```diff
@@ -0,0 +1,35 @@
+import styled from "styled-components";
+import styles from "@/styles";
+
+export default function PreferencesButton({
+  value,
+  selectedPreferences,
+  setPreferences,
+}) {
+  function handleClick() {
+    if (selectedPreferences.includes(value)) {
+      setPreferences(
+        selectedPreferences.filter((preference) => preference !== value)
+      );
+    } else {
+      setPreferences([...selectedPreferences, value]);
+    }
+  }
+  return (
+    <Button
+      onClick={handleClick}
+      selected={selectedPreferences.includes(value)}
+    >
+      {value}
+    </Button>
+  );
+}
```

Comment:
I am a little bit confused about what you are trying to do here. The handle click function is to determine whether the button was selected or not. But then you are not working with this information any further. What happens with the "selected" prop? You need to pass it to the form in order to derive any form of action from it.

I would also recommend maintaining the component as "dumb" as possible. meaning try to deal with most of your logic on the page or more overarching components. But here is a suggestion for a bit more readability in your approach:

```suggestion
export default function PreferencesButton({
  value,
  preference
}) 
const [isPreference, setIsPreference] = useState(false);

function togglePreference() {
setPreference(!preference);
}

  return (
    <Button
      onClick={togglePreference}
    >
      {value}
    </Button>
  );
}
```

Reply:
(don't forget to import useState if you do this)

Reply:
I didnt know i could use the buttons like checkboxes. What I was hoping to do, was to create a reusable button for all the different types of preferences (like company, weather, etc., see US3+4). So that when you have let's say a) a list of possible locations to select, and b) state of each of selectedLocations on the parent component (PreferencesForm), I can use .map to create a set of PreferenceButtons for these options. Inside of the PreferenceButtons I only check whether the option is in the selectedLocations array or not. And when the user clicks the button the handleClick function is fired that takes in the state and setter of the selectedLocations (or selectedCompany, or selectedWeather, etc.) to add or resp. remove the value from this list - that's why this handleClick function exists on the PreferencesButton. The filtering then will happen on the parent component by clicking some sort of submit button that I still need to create(US3).
I didn't want to use checkboxes or radio buttons, so I thought having an array for each category was an okay way🤔  Also I wanted to keep the selected options, hence the lifting of the state of the preferences(?). (Wow, sorry for the novel).

Reply:
Oh and in the next [US](https://github.com/users/izabelladobielewska/projects/2/views/2?pane=issue&itemId=36014740) I got the "show matching games" button (like a submit button?) to filter the cards. So I would just implement this part of the next user story already now. I hope it's okay?